### PR TITLE
Exclude startup, shutdown, etc from being considered user-level tests

### PR DESCRIPTION
**Pull request by decibel**
Adds an exclude argument to findfuncs(). This change breaks existing usage of findfuncs because (name,...) vs (text,...) now matters, but since it's not an actual tap function I think that's OK.

It would be nice to also add this capability to runtests() but that would break a TAP function. Though, we could have the text version first see if $1 matches the name of a schema and if it does treat it as such; not sure if that's worth doing or not. It would also make it confusing if you typo'd the schema name without casting to name; you wouldn't get an error just a bizarre lack of tests running.

### CODE REVIEW

File: sql/pgtap.sql.in [hunk 1]
```diff
@@ -5924,11 +5924,16 @@ RETURNS TEXT[] AS $$
           JOIN pg_catalog.pg_namespace n ON p.pronamespace = n.oid
          WHERE n.nspname = $1
            AND p.proname ~ $2
+           AND ($3 IS NULL OR p.proname !~ $3)
```
Comment (theory) [inline sql/pgtap.sql.in]: Does this work when `$3` is `NULL`? IME, I've had to use `CASE WHEN` to keep it from evaluating both sides of the `OR` statement when `$3` is `NULL`. Something like:

    AND CASE WHEN $3 IS NULL THEN TRUE ELSE p.proname !~ $3 END

**Comment by theory** [comment]
So I guess the point of this change is to prevent `findfuncs()` from finding startup/setup/teardown/shutdown tests? Makes sense.

File: sql/pgtap.sql.in (hunk 1 above)
Comment (decibel) [inline sql/pgtap.sql.in]: >> +           AND ($3 IS NULL OR p.proname !~ $3)
>
> Does this work when |$3| is |NULL|? IME, I've had to use |CASE WHEN| to
> keep it from evaluating both sides of the |OR| statement when |$3| is
> |NULL|. Something like:
>
> |AND CASE WHEN $3 IS NULL THEN TRUE ELSE p.proname !~ $3 END

Dunno. Logically it doesn't matter. nameregexne() (which implements !~) 
is strict, so it won't even be called. I doubt CASE is faster than OR, 
and it'll certainly take a lot longer to parse.

**Comment by theory** [comment]
Would you update the docs and the upgrade script, Jim?

**Comment by decibel** [comment]
Upgrade script added, though I didn't change any of the version numbers. Wasn't sure when you typically did that. I assumed next version would be 0.96.0...

Also didn't make any doc changes; ISTM this is the behavior that the docs already implied for runtest(); were there specific changes you were thinking of?

**Comment by theory** [comment]
Looks good to me. You already added the proper args to `runtests()`. I guess it'd make sense to allow other filters there? Maybe `do_tap()`, too?

**Comment by decibel** [comment]
What other filters did you have in mind? ISTM if you needed filtering at that level you'd handle it through the regex that's already supported...

**Comment by theory** [comment]
@decibel `do_tap()` just passes off to `findfuncs()`, just like `runtests()` does.

**Comment by decibel** [comment]
Sorry, I'm not really groking what you're trying to say from your comment a few days ago.
I don't see any reason to include do_tap in the changes already made, because it doesn't do anything with setup/teardown functions.

OTOH, if you're saying it would be nice to expose the extra filtering option to do_tap() and runtests() then we could do that.

**Comment by theory** [comment]
Yeah, I was responding to this comment from your OP:

> It would be nice to also add this capability to runtests() but that would break a TAP function. Though, we could have the text version first see if $1 matches the name of a schema and if it does treat it as such; not sure if that's worth doing or not. It would also make it confusing if you typo'd the schema name without casting to name; you wouldn't get an error just a bizarre lack of tests running.

**Comment by decibel** [comment]
Ahh, yeah. Now I remember thinking about that as I was writing the patch.

**Pull request merged**